src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.Validation.cs

```diff
@@ -69,15 +70,17 @@ internal static unsafe partial class Utf16Utility
             {
                 if (inputLength >= Vector128<ushort>.Count)
                 {
-                    Vector128<ushort> vector0080 = Vector128.Create((ushort)0x80);
-                    Vector128<ushort> vectorA800 = Vector128.Create((ushort)0xA800);
-                    Vector128<short> vector8800 = Vector128.Create(unchecked((short)0x8800));
-                    Vector128<ushort> vectorZero = Vector128<ushort>.Zero;
+                    Vector128<ushort> vector0080 = Vector128.Create((ushort)0x0080);
+                    Vector128<ushort> vector7800 = Vector128.Create((ushort)0x7800);
+                    Vector128<ushort> vectorA000 = Vector128.Create((ushort)0xA000);
 
                     Vector128<byte> bitMask128 = BitConverter.IsLittleEndian ?
```

Comment:
This code is here already, but why?

`Sse.IsSupported` implies little endian and the commented out condition in L69 checks for LE too (and ARM is XOR to x86).
So this could be simplified. 

https://github.com/dotnet/runtime/pull/39652 introduced this.

---

GH doesn't let me comment there, but should L38 be moved below the check in L41?

Reply:
I think the idea is that ARM support isn't supposed to be predicated on little-endian, so the `&& IsLittleEndian` check at the top of the method shouldn't really exist. But because we didn't have test hardware available we couldn't actually exercise these code paths. I agree that ideally this would all get removed. But, eh, it's not hurting anything by being there right now, and it probably makes more sense to clean it up once ARM vectorization support gets reintroduced.